### PR TITLE
feat(/tag): Support for edits and deleting

### PR DESCRIPTION
**Pull request by Zickles**
allows you to delete tags 
allows you to edit tags through a modal

### CODE REVIEW

File: src/events/interactionCreate.ts [hunk 1]
```diff
@@ -87,6 +87,25 @@ export const execute = async (interaction: Interaction) => {
           .setDescription(`The tag \`${name}\` has been added successfully`);
         await interaction.reply({ embeds: [embed], ephemeral: true });
       }
+      if (interaction.customId === 'tagEditForm') {
```
Comment (Kathund) [inline src/events/interactionCreate.ts]: Add a check here for if the user still has permissions

File: src/commands/tag.ts [hunk 2]
```diff
@@ -90,6 +107,60 @@ export const execute = async (interaction: ChatInputCommandInteraction) => {
         }
         break;
       }
+      case 'edit': {
+        if (memberRoles.some((role) => [contributorsRole, teamRole, devRole].includes(role))) {
```
Comment (Kathund) [inline src/commands/tag.ts]: Team and dev only please

File: src/commands/tag.ts [hunk 3]
```diff
@@ -90,6 +107,60 @@ export const execute = async (interaction: ChatInputCommandInteraction) => {
...
+        } else {
+          return await interaction.reply({
+            content: 'You do not have permission to use this command',
+            ephemeral: true,
+          });
+        }
+        break;
+      }
+      case 'delete': {
+        if (memberRoles.some((role) => [contributorsRole, teamRole, devRole].includes(role))) {
```
Comment (Kathund) [inline src/commands/tag.ts]: Team and dev only please

**Review by Kathund — CHANGES_REQUESTED** [review]

File: src/events/interactionCreate.ts (hunk 1 above)
Comment (Zickles) [reply src/events/interactionCreate.ts]: :+1:

File: src/commands/tag.ts (hunk 2 above)
Comment (Zickles) [reply src/commands/tag.ts]: :+1:

File: src/commands/tag.ts (hunk 3 above)
Comment (Zickles) [reply src/commands/tag.ts]: :+1:

File: src/events/interactionCreate.ts [hunk 4]
```diff
@@ -87,6 +89,26 @@ export const execute = async (interaction: Interaction) => {
...
+        if (memberRoles.some((role) => [teamRole, devRole].includes(role))) return
+        const name = interaction.fields.getTextInputValue('tagFormUpdatedName').toLowerCase();
```
Comment (Kathund) [inline src/events/interactionCreate.ts]: Is it possible to find a different way to pass in the name? I dont want people to be able to change the name of the tag. This "might" also cause an issue where if the name isnt already in the database it willl throw a random error and **"maybe"** crash the bot.

In theory have them select the name in the command options first then have it check if it exists before letting them edit the infomation

**Review by Kathund — CHANGES_REQUESTED** [review]

File: src/events/interactionCreate.ts (hunk 4 above)
Comment (Zickles) [reply src/events/interactionCreate.ts]: Yeah I was going to do it like that and then I didn't and idk why  but yeah I can do it like that

**Review by Kathund — APPROVED** [review]

**Pull request merged**